server/odc-service/src/main/java/com/oceanbase/odc/metadb/regulation/ruleset/RuleApplyingRepository.java

```diff
@@ -43,5 +47,30 @@ Optional<RuleApplyingEntity> findByOrganizationIdAndRulesetIdAndRuleMetadataId(L
     List<RuleApplyingEntity> findByOrganizationIdAndRuleMetadataName(@Param("organizationId") Long organizationId,
             @Param("name") String name);
 
+    @Transactional
+    int deleteByOrganizationIdAndRulesetId(Long organizationId, Long rulesetId);
+
+
+    default List<RuleApplyingEntity> batchCreate(List<RuleApplyingEntity> entities) {
+        String sql = InsertSqlTemplateBuilder.from("regulation_rule_applying")
+                .field(RuleApplyingEntity_.ORGANIZATION_ID)
+                .field(RuleApplyingEntity_.RULESET_ID)
+                .field(RuleApplyingEntity_.RULE_METADATA_ID)
+                .field(RuleApplyingEntity_.APPLIED_DIALECT_TYPES)
+                .field(RuleApplyingEntity_.ENABLED)
+                .field(RuleApplyingEntity_.LEVEL)
+                .field(RuleApplyingEntity_.PROPERTIES_JSON)
+                .build();
+        List<Function<RuleApplyingEntity, Object>> getter = valueGetterBuilder().add(
+                RuleApplyingEntity::getOrganizationId)
+                .add(RuleApplyingEntity::getRulesetId)
+                .add(RuleApplyingEntity::getRuleMetadataId)
+                .add(RuleApplyingEntity::getAppliedDialectTypes)
```

Comment:
The `appliedDialectTypes` is a List, I am not sure whether the `batchCreate` method support this. You better to test that.

Reply:
Yes you are right, we need to convert the list to a string. Thanks for your kind reminder:)